lib/src/native/realm_core.dart

```diff
@@ -1738,10 +1749,14 @@ class ReleasableHandle<T extends NativeType> extends HandleBase<T> {
     if (released) {
       return;
     }
-
-    _realmLib.realm_dart_delete_finalizable(_finalizableHandle, this);
+    _nativeFinalizer.detach(this);
     _realmLib.realm_release(_pointer.cast());
     released = true;
+    assert(() {
```

Comment:
we can remove these per above

Reply:
asserts are automatically dropped in release mode. I find it convenient to have the ability to get trace from the dart side as well.

Reply:
Reworked as discussed on round-table meeting